sdk/storage/azure-storage-blob/tests/test_common_blob.py

```diff
@@ -2349,4 +2397,18 @@ def test_set_blob_tier_for_a_version(self, resource_group, location, storage_acc
         self.assertEqual(props.blob_tier, 'Hot')
         self.assertEqual(origin_props.blob_tier, 'Cool')
 
-#------------------------------------------------------------------------------
+    @GlobalStorageAccountPreparer()
+    def test_access_token_refresh_after_retry(self, resource_group, location, storage_account, storage_account_key):
+        def fail_response(response):
+            response.http_response.status_code = 408
+
+        cred_token = MockCredential()
+        cred_policy = CredentialRotationPolicy(cred_token)
+        self._setup(storage_account, storage_account_key, [cred_policy], total_retries=2)
```

Comment:
This is overall good progress, but I think this test doesn't assert sequence between retry and auth policy.
What it test currently is that "additional policy" is called twice, we need to assert that implicit policies created by pipeline builder are in right order. 

Instead of passing storage_account_key we should pass custom TokenCredential that does more less what cred_policy does right now.

Reply:
I actually found a much simpler approach using `generate_fake_token()`. It's used as a credential and it achieves the test we have been trying to do :)